### PR TITLE
Gp function

### PR DESCRIPTION
**Pull request by LauraBreimann**

### CODE REVIEW

File: gunpowder_augmentor.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import gunpowder as gp
+import math
+import numpy as np
+import matplotlib.pyplot as plt
+import torch
+
+
+def process_zarr_data(load_path, output_shape=(30, 128, 128), stack_size=5,  device='cuda'):
+    # Open the zarr file
+    f = zarr.open(load_path, 'r')
```
Comment (msschwartz21) [inline gunpowder_augmentor.py]: Not used

File: gunpowder_augmentor.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    # Build the pipeline and request the batch
+    with gp.build(pipeline):
+        batch = pipeline.request_batch(request)
+
+    # Move the batch data to the specified device
+    if device == 'cuda':
+        batch[raw].data = torch.from_numpy(batch[raw].data).cuda()
+        batch[gt].data = torch.from_numpy(batch[gt].data).cuda()
+
+    return batch
```
Comment (msschwartz21) [inline gunpowder_augmentor.py]: Return `pipeline` before building and the `request`

User calls outside of this function 
```
with gp.build(pipeline):
        batch = pipeline.request_batch(request)
```

File: gunpowder_augmentor.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
+import zarr
+import gunpowder as gp
+import math
+import numpy as np
+import matplotlib.pyplot as plt
+import torch
+
+
+def process_zarr_data(load_path, output_shape=(30, 128, 128), stack_size=5,  device='cuda'):
```
Comment (msschwartz21) [inline gunpowder_augmentor.py]: Consider renaming function to `prepare_gunpowder_pipeline`

File: gunpowder_augmentor.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    simple_augment = gp.SimpleAugment(
+    mirror_probs = [0,0,0],
+    transpose_probs = [0,0,0])
+    
+    elastic_augment = gp.ElasticAugment(
+    control_point_spacing=(30, 30, 30),
+    jitter_sigma=(1.0, 1.0, 1.0),
+    rotation_interval=(0, math.pi/2),
+    spatial_dims = 3
+    )
```
Comment (msschwartz21) [inline gunpowder_augmentor.py]:
```suggestion
    elastic_augment = gp.ElasticAugment(
        control_point_spacing=(30, 30, 30),
        jitter_sigma=(1.0, 1.0, 1.0),
        rotation_interval=(0, math.pi/2),
        spatial_dims = 3
    )
```
Try `black` for formatting 
`ruff` for linting

File: gunpowder_augmentor.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    raw,
+    scale_min=0.9,
+    scale_max=1.1,
+    shift_min=-0.01,
+    shift_max=0.01)
+
+    noise_augment = gp.NoiseAugment(raw, mode='poisson')
+
+    normalize_raw = gp.Normalize(raw)
+    normalize_gt = gp.Normalize(gt, factor=1.0/255.0)
```
Comment (msschwartz21) [inline gunpowder_augmentor.py]: Delete norm gt

File: gunpowder_augmentor.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+
+    ##### Gunpowder pipeline #####
+
+    # Load the data
+    source = tuple(gp.ZarrSource(
+    load_path,
```
Comment (msschwartz21) [inline gunpowder_augmentor.py]: Improve generalizability by iterating over list of paths and fov 

Think about data structure for this input

**Review by msschwartz21 — CHANGES_REQUESTED** [review]

File: gunpowder_augmentor.py (hunk 1 above)
Comment (LauraBreimann) [reply gunpowder_augmentor.py]: Removed it!

File: gunpowder_augmentor.py (hunk 4 above)
Comment (LauraBreimann) [reply gunpowder_augmentor.py]: thanks, changed it (hopefully)

File: gunpowder_augmentor.py (hunk 2 above)
Comment (LauraBreimann) [reply gunpowder_augmentor.py]: returning pipeline and request, calling outside

File: gunpowder_augmentor.py (hunk 3 above)
Comment (LauraBreimann) [reply gunpowder_augmentor.py]: renamed!

File: gunpowder_augmentor.py (hunk 5 above)
Comment (LauraBreimann) [reply gunpowder_augmentor.py]: deleted

File: gunpowder_augmentor.py (hunk 6 above)
Comment (LauraBreimann) [reply gunpowder_augmentor.py]: now loads a list of images

**Comment by LauraBreimann** [comment]
new version of the gunpowder script

**Review by stemarcotti — APPROVED** [review]

**Pull request merged**